cmd/builder/internal/builder/main.go

```diff
@@ -29,14 +29,34 @@ var (
 	skipStrictMsg      = "Use --skip-strict-versioning to temporarily disable this check. This flag will be removed in a future minor version"
 )
 
-func runGoCommand(cfg Config, args ...string) ([]byte, error) {
+func getGoEnv(cfg Config) ([]string, error) {
+	// #nosec G204 -- cfg.Distribution.Go is trusted to be a safe path and the caller is assumed to have carried out necessary input validation
+	cmd := exec.Command(cfg.Distribution.Go, "env")
+	cmd.Dir = cfg.Distribution.OutputPath
+
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	if err := cmd.Run(); err != nil {
+		return nil, fmt.Errorf("go env failed: %w, error message: %s", err, stderr.String())
+	}
+
+	str := stdout.String()
+	str = strings.ReplaceAll(str, "'", "")
+
+	return strings.Split(str, "\n"), nil
+}
+
+func runGoCommand(cfg Config, env []string, args ...string) ([]byte, error) {
```

Comment:
I'm not a huge fan of the new function signature considering `env` won't often be used. I think if we add an option to pass env vars to the go build command, we may be able to just read from `cfg` and eliminate this.